scripts/issue/474/upgrade_arb_crv_univ3.py

```diff
@@ -0,0 +1,128 @@
+from decimal import Decimal
+
+from brownie import accounts, web3, interface, chain
+from rich.console import Console
+
+from great_ape_safe import GreatApeSafe
+from helpers.addresses import ADDRESSES_ARBITRUM
+from helpers.addresses import registry
+
+console = Console()
+
+DEV_MULTI = registry.arbitrum.badger_wallets.dev_multisig
+PROXY_ADMIN = registry.arbitrum.proxyAdminDev
+
+TRICRYPTO_STRATEGY = ADDRESSES_ARBITRUM["strategies"]["native.tricrypto"]
+TRICRYPTO_LOGIC = ADDRESSES_ARBITRUM["logic"]["native.tricrypto"]
+TRICRYPTO_VAULT = ADDRESSES_ARBITRUM["sett_vaults"]["bcrvTricrypto"]
+
+RENCRV_STRATEGY = ADDRESSES_ARBITRUM["strategies"]["native.renCrv"]
+RENCRV_LOGIC = ADDRESSES_ARBITRUM["logic"]["native.renCrv"]
+RENCRV_VAULT = ADDRESSES_ARBITRUM["sett_vaults"]["bcrvRenBTC"]
+
+UNIV3_ROUTER = "0xE592427A0AEce92De3Edee1F18E0157C05861564"
+
+
+def main(simulation=True):
+    safe = GreatApeSafe(DEV_MULTI)
+
+    console.print(
+        f"Updating Tricrypto strategy logic to {TRICRYPTO_LOGIC}"
+    )
+    upgrade_strategy_logic(
+        TRICRYPTO_STRATEGY,
+        TRICRYPTO_LOGIC,
+        TRICRYPTO_VAULT,
+        safe,
+        simulation,
+    )
+
+    console.print(f"Updating Rencrv strategy logic to {RENCRV_LOGIC}")
+    upgrade_strategy_logic(
+        RENCRV_STRATEGY, RENCRV_LOGIC, RENCRV_VAULT, safe, simulation
+    )
+
+    safe.post_safe_tx(call_trace=True)
+
+
+def upgrade_strategy_logic(
+    strategy_proxy_address: str,
+    logic_address: str,
+    vault_proxy_address: str,
+    safe: GreatApeSafe,
+    simulation: bool = True,
+):
+    vault = interface.ITheVault(vault_proxy_address, owner=safe.account)
+    proxy_admin = interface.IProxyAdmin(PROXY_ADMIN, owner=safe.account)
+    strat_proxy = interface.ICrvStrategy(strategy_proxy_address, owner=safe.account)
+    controller_proxy = interface.IController(strat_proxy.controller(), owner=vault)
+
+    prev_strategist = strat_proxy.strategist()
+    prev_controller = strat_proxy.controller()
+    prev_gov = strat_proxy.governance()
+    prev_guardian = strat_proxy.guardian()
+    prev_keeper = strat_proxy.keeper()
+    prev_perFeeG = strat_proxy.performanceFeeGovernance()
+    prev_perFeeS = strat_proxy.performanceFeeStrategist()
+    prev_reward = strat_proxy.reward()
+    prev_unit = strat_proxy.uniswap()
+    prev_gauge = strat_proxy.gauge()
+    prev_gauge_factory = strat_proxy.gaugeFactory()
+    prev_swapr_router = strat_proxy.SWAPR_ROUTER()
+
+    if simulation:
+        # Harvest to clear any pending rewards for fresh test case
+        gauge = strat_proxy.gauge()
+        strat_proxy.harvest()
+
+        # Harvest on old strat, store gain in want
+        want = interface.IERC20(gauge)
```

Comment:
better to use `interface.ERC20` or `safe.contract`, `IERC20` is an incomplete interface and initialising an address with it can give problems on your local later on.